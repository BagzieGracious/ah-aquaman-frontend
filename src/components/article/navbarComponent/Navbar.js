import React, { Component } from "react";
import { Link } from "react-router-dom";
class Navbar extends Component {
    state = {
      hidden: false,
    }

    componentWillMount() {
      if (localStorage.getItem("token")) {
        this.setState({ hidden: true });
      }
    }

    render() {
      const title = "Author's Haven";
      let image = "";
     
      if (!this.state.hidden) {
        return (
          <header>
            <h1 className="header-title">{title}</h1>
            <nav className="navbar navbar-expand-lg navbar-light">
              <div className="container">
                <div className="collapse navbar-collapse offset">
                  <ul className="nav navbar-nav">
                    <li className="nav-item active"><Link to="/">Articles</Link></li>
                  </ul>
                  <ul className="nav navbar-nav navbar-right ml-auto">
                    <li className="nav-item">
                      <Link to="/login" className="create">Login </Link>
                    </li>
                  </ul>
                </div>
              </div>
            </nav>
          </header>
        );
      }
      
      return (
        <header>
          <h1 className="header-title">{title}</h1>
          <nav className="navbar navbar-expand-lg navbar-light">
            <div className="container">
              <div className="collapse navbar-collapse offset">
                <ul className="nav navbar-nav">
                  <li className="nav-item active"><Link to="/">Articles</Link></li>
                </ul>
                <ul className="nav navbar-nav navbar-right ml-auto">
                  {this.state.hidden && (
                    <Link to="/profile">
                      <li className="nav-item dropdown">
                        <img src={image} alt="" className="img-responsive profile-img" />
                      </li>
                    </Link>
                  )}
                  <li className="nav-item">
                    {this.state.hidden && (
                      <Link to="/create_article" className="create">
                        Create
                      </Link>
                    )}
                  </li>
                  <li className="nav-item">
                    {!this.state.hidden && <Link to="/signup" className="create">Register </Link>}
                  </li>
                  <li className="nav-item">
                    {!this.state.hidden && <Link to="/login" className="create">Login </Link>}
                  </li>
                  <li className="nav-item">
                    {this.state.hidden && <Link to="/logout" className="create">Logout </Link>}
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </header>
      );
    }
}



export default Navbar;
