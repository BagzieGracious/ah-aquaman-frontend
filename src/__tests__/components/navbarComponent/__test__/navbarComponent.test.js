import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Navbar from "../../../../components/article/navbarComponent/Navbar";

const initialState = {};
const mockStore = configureStore();
const store = mockStore(initialState);

describe("NavBar not logged in test", () => {
  const props = {
    state: {
      hidden: true,
    },
  };

  it("should render without crashing", () => {
    const wrapper = shallow(<Navbar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
