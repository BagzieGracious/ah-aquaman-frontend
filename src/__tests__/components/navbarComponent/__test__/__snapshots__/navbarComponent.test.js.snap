// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar not logged in test should render without crashing 1`] = `ShallowWrapper {}`;
